backend/models/user.js

```diff
@@ -240,126 +237,125 @@ schema.statics.updatePassword = function(logger, username, oldPassword, token, n
 schema.statics.usernameRegExp = /^[a-zA-Z][\w]{1,63}$/;
 
 schema.statics.createUser = function(logger, username, password, customData, tokenExpiryTime, skipCheckEmail){
-	"use strict";
-	return ModelFactory.dbManager.getAuthDB().then(adminDB => {
 
-		let cleanedCustomData = {};
-		let emailRegex = /^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/;
+	if (customData) {
+		return ModelFactory.dbManager.getAuthDB().then(adminDB => {
 
-		if(config.auth.allowPlusSignInEmail){
-			emailRegex = /^([+a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/;
-		}
-
-		if(customData && (!customData.email || !customData.email.match(emailRegex))){
-			return Promise.reject({ resCode: responseCodes.SIGN_UP_INVALID_EMAIL });
-		}
+			let cleanedCustomData = {};
+			let emailRegex = /^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/;
 
+			if (config.auth.allowPlusSignInEmail) {
+				emailRegex = /^([+a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/;
+			}
 
-		if(!this.usernameRegExp.test(username)){
-			return Promise.reject({ resCode: responseCodes.INVALID_USERNAME});
-		}
+			if (!customData.email || !customData.email.match(emailRegex)) {
+				return Promise.reject({ resCode: responseCodes.SIGN_UP_INVALID_EMAIL });
+			}
 
-		for(let i=0 ; i < C.REPO_BLACKLIST_USERNAME.length; i++){
-			if(C.REPO_BLACKLIST_USERNAME[i] === username){
-				return Promise.reject({ resCode: responseCodes.INVALID_USERNAME });
+			if (!this.usernameRegExp.test(username)) {
+				return Promise.reject({ resCode: responseCodes.INVALID_USERNAME});
 			}
-		}
-	
 
-		["firstName", "lastName", "email", "mailListOptOut"].forEach(key => {
-			if (customData && customData[key]){
-				cleanedCustomData[key] = customData[key];
+			for (let i = 0; i < C.REPO_BLACKLIST_USERNAME.length; i++) {
```

Comment:
should probably convert new username to lower case (the list is in constant is all in lower case), and just do indexOf or something similar instead of looping